pkg/apis/kubeless/v1beta1/http_trigger.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright (c) 2016-2017 Bitnami
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// HTTPTrigger object
+type HTTPTrigger struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              HTTPTriggerSpec `json:"spec"`
+}
+
+// HTTPTriggerSpec contains func specification
+type HTTPTriggerSpec struct {
+	FunctionName  string         `json:"function-name"` // Name of the associated function
+	ServiceSpec   v1.ServiceSpec `json:"service"`
```

Comment:
Yes but some people need the flexibility to define some annotations in order to use some external services ([slack link](https://kubernetes.slack.com/archives/C92FRH92S/p1520455559000346)). Also note that this would be for advanced use only, the normal usage would be using the kubeless CLI with the flags `--port` `--hostname`...